File: im-sdk-java/src/com/farsunset/cim/sdk/client/model/HeartbeatRequest.java

package com.farsunset.cim.sdk.client.model;

import java.io.Serializable;

import com.farsunset.cim.sdk.client.constant.CIMConstant;

/**
 * 服务端心跳请求
 *
 */
public class HeartbeatRequest implements Serializable, Protobufable {

	private static final long serialVersionUID = 1L;
	private static final String TAG = "SERVER_HEARTBEAT_REQUEST";
	private static final String CMD_HEARTBEAT_RESPONSE = "SR";

	private static HeartbeatRequest object = new HeartbeatRequest();

	private HeartbeatRequest() {

	}

	public static HeartbeatRequest getInstance() {
		return object;
	}

	@Override
	public byte[] getByteArray() {
		return CMD_HEARTBEAT_RESPONSE.getBytes();
	}

	public String toString() {
		return TAG;
	}

	@Override
	public byte getType() {
		return CIMConstant.ProtobufType.S_H_RQ;
	}

}

File: im-sdk-java/src/vip/qsos/im/lib/client/model/HeartbeatRequest.kt
package vip.qsos.im.lib.client.model

import vip.qsos.im.lib.client.constant.IMConstant

/**
 * @author : 华清松
 * 服务端心跳请求实体,客户端应发送一条心跳消息给服务器
 */
class HeartbeatRequest private constructor() : IProtobufAble {

    companion object {
        private const val serialVersionUID = 1L
        const val TAG = "SERVER_HEARTBEAT_REQUEST"
        const val SERVER_HEARTBEAT_REQUEST = "SR"
        val instance = HeartbeatRequest()
    }

    override val byteArray: ByteArray
        get() = SERVER_HEARTBEAT_REQUEST.toByteArray()

    override val type: Byte
        get() = IMConstant.ProtobufType.HEART_RQ

    override fun toString(): String {
        return TAG
    }

}